### DISCUSSION TITLE
Question about DTLS verification

### DISCUSSION THREAD

**Discussion by UkuKert** [OvenMediaLabs/OvenMediaEngine#1636] [Help]
Hello!

I want to start with a big thanks to you guys for this amazing tool. I love using OME, because it is so easy to use and it just works!

I was digging around the WebRTC implementation and found this piece in the DTLS code:
https://github.com/AirenSoft/OvenMediaEngine/blob/32e2ebc3b033564a83ac93cc77a213e60ecba192/src/projects/modules/dtls_srtp/dtls_transport.cpp#L178C1-L184C2

Do I understand correctly that currently incoming DTLS packets signature is not verified at this moment? As far as I understand this means that incoming WebRTC packets are not validated. Is this correct or am I mistaken?

A follow-up question is that are the outgoing WebRTC packets encrypted?

I apologize if my questions are stupid. I'm quite fresh into WebRTC and may not yet understand all the intricacies well.

Thank you for your time and contributions,
Uku

**Comment by getroot** [comment]
Verifying the certificate is not strictly necessary. All sent and received packets are SRTP encrypted with keys exchanged via DTLS.
